### DISCUSSION TITLE
Multi-Factor Authentication for Admin Panel [Dashboard]

### DISCUSSION THREAD

**Discussion by dschnabel** [supabase/supabase#12761] [Feature Requests]
### Update: [this is now implemented and available on supabase.com](https://github.com/orgs/supabase/discussions/12761#discussioncomment-7309367)

---

It seems that MFA is currently not possible for the admin panel (https://app.supabase.com/), only for projects. Securing the admin panel with MFA seems like an important feature to have, since an attacker could gain access to all the admin's projects and users data.

Are there any plans to add MFA to the admin panel?

**Comment by BigDarkAkula** [comment]
This seems like pretty serious vulnerability that will prohibit many users from being able to adopt the platform.

**Comment by J0** [comment]
Hey @dschnabel, @BigDarkAkula
 
Thanks for our feedback! Adding MFA to our platform is on our roadmap - do stay tuned!

Jo

**Reply by rogerclark** [reply]
Any updates on this? This seems really important. Thanks!

**Reply by ryanparks** [reply]
Same boat -- our team is hoping to deploy a product with Supabase under the hood, but this gives us pause.

**Reply by BigDarkAkula** [reply]
Would self hosting be the alternative here to make sure your supabase instance secure from a brute force or some other type of attack? I REALLY don't want to do that.  It would be really nice to know what the time horizon looks like to implement this type of functionality.

**Reply by rohovskoi** [reply]
Hi @J0, is there an update on 2MFA for the admin dashboard? That's a significant roadblock for us.

**Comment by lazarette** [comment]
Agree - nothing secure can be hosted on supabase when no data hosted here has mfa protection... I'd like to see MFA and IP restrictions on the admin panel.

**Comment by lukereichold** [comment]
Would pay for this! The lack of this feature is currently a dealbreaker for using Supabase for client projects 😞

**Comment by carlosfarfangalindoNL222** [comment]
Please add this feature, this is really a big vulnerability!

**Comment by dschnabel** [comment]
@J0 any progress on this?

**Reply by sigma-andex** [reply]
According to @awalias it's on the way..

**Comment by J0** [comment]
Hey all,

Thanks for waiting - MFA is available on the dashboard now. Do read the [docs around it](https://supabase.com/docs/guides/platform/multi-factor-authentication) here. You can enroll an MFA (TOTP) factor on [the security page](https://supabase.com/dashboard/account/security)

Going to close the thread but we will continue to take feedback in this discussion thread. Alternatively, feel free to open a separate thread as well.

Thanks!
